surfactant/infoextractors/native_lib_file.py

```diff
@@ -0,0 +1,79 @@
+import json
+import pathlib
+import re
+from typing import Any, Dict, List
+
+from loguru import logger
+
+import surfactant.plugin
+from surfactant.sbomtypes import SBOM, Software
+
+
+def supports_file(filetype) -> bool:
+    return filetype in ("PE", "ELF", "MACHOFAT", "MACHOFAT64", "MACHO32", "MACHO64")
+
+
+@surfactant.plugin.hookimpl
+def extract_file_info(sbom: SBOM, software: Software, filename: str, filetype: str) -> object:
+    if not supports_file(filetype):
+        return None
+    return extract_native_lib_info(filename)
+
+
+def extract_native_lib_info(filename):
+    native_lib_info: Dict[str, Any] = {"nativeLibraries": []}
+    native_lib_patterns = pathlib.Path(__file__).parent / "native_lib_patterns.json"
+
+    # Load regex patterns into database var
+    try:
+        with open(native_lib_patterns, "r") as regex:
+            database = json.load(regex)
+    except FileNotFoundError:
+        logger.warning(f"File not found: {native_lib_patterns}")
+        return None
+
+    found_libraries = set()
+
+    # Match based on filename
+    filenames_list = match_by_attribute("filename", filename, database)
+    if len(filenames_list) > 0:
+        for match in filenames_list:
+            library_name = match["library"]
+            if library_name not in found_libraries:
+                native_lib_info["nativeLibraries"].append(match)
```

Comment:
I'm wondering if this should use a different key like "isNativeLibrary" -- theoretically a file name matching would imply that the file itself is that library/binary.

e.g. A file could have a copy of OpenSSL embedded in it (contains the native library based on a pattern match), but if the file name matches "libopenssl" it would be better described as "being" OpenSSL (which is a more accurate description than just saying it "contains" OpenSSL). It is also unlikely that a file will match patterns for multiple different things (if it does, then those file name patterns are probably not specific enough).